hips/tiles/description.py

```diff
@@ -0,0 +1,99 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+"""Classes and functions to manage HiPS tiles."""
+
+
+from collections import OrderedDict
+
+__all__ = [
+    'HipsDescription',
+]
+
+class HipsDescription:
+    """This class provides methods for parsing the HiPS properties file.
+
+    It also provides multiple getters around the commonly used HiPS properties.
+
+    Parameters
+    ----------
+    properties : str
+        A string containing HiPS tile properties
+    """
+
+    def __init__(self, properties: str) -> None:
+        super(HipsDescription, self).__init__()
+        self.properties = self.parse_file_properties(properties)
+
+    @classmethod
+    def read_file(cls, filename: str) -> OrderedDict:
+        """Reads text from a file and returns an ordered dictionary containing the HiPS file properties.
+
+        This method reads a file given by the parameter and returns an ordered dictionary
+        containing the HiPS file properties by calling the `parse_file_properties` method.
+
+        Parameters
+        ----------
+        filename : str
+            Name of the file containing HiPS file properties
+
+        Returns
+        -------
+        dict_properties : OrderedDict
+        """
+        file = open(filename, 'r')
```

Comment:
Always use this pattern to read (and write) files (to make sure they are closed immediately after the reading is done, in a predictable manner:
```
with open(filename) as fh:
    text = fh.read()
return cls.parse(text)
```